client.go

```diff
@@ -49,22 +51,29 @@ func New(cfg Configuration) *Client {
 		globalTags: cfg.Tags,
 	}
 
-	if cfg.FlushInterval > 0 {
+	if cfg.FlushInterval > 0 && cfg.AutoFlush {
 		statful.StartFlushInterval(cfg.FlushInterval)
 	}
 
 	return statful
 }
 
-// Starts a go routine that periodically flushes the metrics of MetricsSender
+// Starts a go routine that periodically flushes the metrics from buffer
+// If AutoFlush is deactivated it just send metrics synchronously.
 // Returns a function that stops the timer.
 func (c *Client) StartFlushInterval(interval time.Duration) {
+	if !c.buffer.autoFlush {
+		c.sendSynchronously()
+		return
+	}
+
```

Comment:
I don't understand this change - I think if the client is configured to not flush automatically and someone calls `StartFlushInterval` ideally we'd return an error saying something like "auto flush is disabled" but to avoid breaking the api I'd just make this a no-op:

```suggestion
	if !c.buffer.autoFlush {
		return
	}
```